code_samples/portable_multiversion_sample.js

```diff
@@ -0,0 +1,224 @@
+    writer.writeUTF('name', this.name);
+    writer.writeInt('age', this.age);
+};
+
+Employee.prototype.getFactoryId = function () {
+    return 1;
+};
+
+Employee.prototype.getClassId = function () {
+    return 1;
+};
+
+Employee.prototype.getVersion = function () {
+    return 1;
+};
+
+//Version 2 Employee class.
+function Employee2(name, age, manager) {
+    this.name = name;
+    this.age = age;
+    this.manager = manager;
+};
+
+Employee2.prototype.readPortable = function (reader) {
+    this.name = reader.readUTF('name');
+    this.age = reader.readInt('age');
+    this.manager = reader.readUTF('manager');
+};
+
+Employee2.prototype.writePortable = function (writer) {
+    writer.writeUTF('name', this.name);
+    writer.writeInt('age', this.age);
+    writer.writeUTF('manager', this.manager);   //specifies version different than the global version.
+};
+
+Employee2.prototype.getFactoryId = function () {
+    return 1;
+};
+
+Employee2.prototype.getClassId = function () {
+    return 1;
+};
+
+Employee2.prototype.getVersion = function () {
+    return 2;
+};
+
+//Version 3 Employee class.
+function Employee3(name, age, manager) {
+    this.name = name;
+    this.age = age;
+    this.manager = manager;
+
+};
+
+Employee3.prototype.readPortable = function (reader) {
+    this.name = reader.readUTF('name');
+    this.age = reader.readUTF('age');
+    this.manager = reader.readUTF('manager');
+};
+
+Employee3.prototype.writePortable = function (writer) {
+    writer.writeUTF('name', this.name);
+    writer.writeUTF('age', this.age);
+    writer.writeUTF('manager', this.manager);
+};
+
+Employee3.prototype.getFactoryId = function () {
+    return 1;
+};
+
+Employee3.prototype.getClassId = function () {
+    return 1;
+};
+
+Employee3.prototype.getVersion = function () {
+    return 3;
+};
+
+
+function PortableFactory() {
+};
+
+PortableFactory.prototype.create = function (classId) {
+    if (classId === 1) {
+        return new Employee();
+    }
+    return null;
+};
+
+function PortableFactory2() {
+};
+
+PortableFactory2.prototype.create = function (classId) {
+    if (classId === 1) {
+        return new Employee2();
+    }
+    return null;
+};
+
+function PortableFactory3() {
+};
+
+PortableFactory3.prototype.create = function (classId) {
+    if (classId === 1) {
+        return new Employee3();
+    }
+    return null;
+};
+
+//Let's now configure 3 clients with 3 different versions of Employee.
+
+var cfg = new Config.ClientConfig();
+cfg.serializationConfig.portableFactories[1] = new PortableFactory();
+
+var cfg2 = new Config2.ClientConfig();
+cfg2.serializationConfig.portableFactories[1] = new PortableFactory2();
+
+var cfg3 = new Config3.ClientConfig();
+cfg3.serializationConfig.portableFactories[1] = new PortableFactory3();
+
+let map, map2, map3;
+let client1,client2,client3;
+
+// Start the Hazelcast Client and connect to an already running Hazelcast Cluster on 127.0.0.1
+Client.newHazelcastClient(cfg).then(function (hz) {
```

Comment:
Improve the readability and include the comments on the python code sample to make it clear what you are doing